dsl_parser/functions.py

```diff
@@ -634,18 +689,47 @@ def evaluate_functions(payload, context,
     return payload
 
 
+def evaluate_capabilities(capabilities,
+                          get_node_instances_method,
+                          get_node_instance_method,
+                          get_node_method,
+                          get_secret_method,
+                          get_capability_method):
+    """Evaluates capabilities definition containing intrinsic functions.
+
+    :param capabilities: The dict of capabilities to evaluate
+    :param get_node_instances_method: A method for getting node instances.
+    :param get_node_instance_method: A method for getting a node instance.
+    :param get_node_method: A method for getting a node.
+    :param get_secret_method: A method for getting a secret.
+    :param get_capability_method: A method for getting a capability.
+    :return: Outputs dict.
+    """
+    capabilities = dict((k, v['value']) for k, v in capabilities.iteritems())
```

Comment:
i'd say let's use `.items()` in new code so that we don't add more things to change when porting to py3

Reply:
also, this always runs on 2.7, right? ie. not on the agent
use a dictcomp if so